src/box/lua/cfg.cc

```diff
@@ -177,9 +180,22 @@ lbox_cfg_set_wal_queue_max_size(struct lua_State *L)
 	return 0;
 }
 
+/**
+ * If set, raise an error on any attempt to set wal_cleanup_delay.
+ */
+static bool wal_cleanup_delay_is_disabled = false;
+TWEAK_BOOL(wal_cleanup_delay_is_disabled);
+
 static int
 lbox_cfg_set_wal_cleanup_delay(struct lua_State *L)
 {
+	if (wal_cleanup_delay_is_disabled) {
+		diag_set(ClientError, ER_DEPRECATED,
+			 "Option wal_cleanup_delay");
+		luaT_error(L);
+	} else {
+		say_warn("Option wal_cleanup_delay is deprecated.");
+	}
```

Comment:
No need to add a tweak to C. We could do something like this right in Lua:

```diff
diff --git a/src/box/lua/load_cfg.lua b/src/box/lua/load_cfg.lua
index 254c0fd83a5c..b7485f221888 100644
--- a/src/box/lua/load_cfg.lua
+++ b/src/box/lua/load_cfg.lua
@@ -504,7 +504,14 @@ local dynamic_cfg = {
     worker_pool_threads     = private.cfg_set_worker_pool_threads,
     -- do nothing, affects new replicas, which query this value on start
     wal_dir_rescan_delay    = nop,
-    wal_cleanup_delay       = private.cfg_set_wal_cleanup_delay,
+    wal_cleanup_delay       = function()
+        if compat... then
+            box.error(...)
+        else
+            log.warn('...')
+        end
+        private.cfg_set_wal_cleanup_delay()
+    end
     wal_retention_period    = private.cfg_set_wal_retention_period,
     custom_proc_title       = function()
         require('title').update(box.cfg.custom_proc_title)
```

Reply:
Good idea, thanks, done.